src/middlewares/Mock.ts

```diff
@@ -0,0 +1,73 @@
+import {IRequest, IResponse} from "../Client";
+import {IMiddleware} from "../Stack";
+import {FetchResponse} from "./Fetch";
+
+export interface IMockHandler<IN, OUT> {
+    match: (options: IN) => boolean|undefined;
+    factory: (options: IN) => OUT;
```

Comment:
Should we rename `factory` to something else? resultFn, resultFactory, .... ? The term factory is biased.